### PR TITLE
1.2.1 proposed

### PR DESCRIPTION
**Pull request by mr-c**
A standing PR to see all the [CWL v1.2.1 proposals](https://github.com/common-workflow-language/cwl-v1.2/issues/54) that have gotten initial approval

- [x] CommandLineTool changelog for v1.2.1
- [x] Workflow / ExpressionTool / Operation changelog for v1.2.1

https://deploy-preview-115--cwl-v1-2-dev.netlify.app/workflow#Changelog_for_v1.2.1

https://deploy-preview-115--cwl-v1-2-dev.netlify.app/commandlinetool#Changelog_for_v1.2.1

### CODE REVIEW

**Review by kinow — DISMISSED** [review]
Thank you!

**Comment by adamnovak** [comment]
I don't see anything here that raises any alarm bells for Toil.

**Review by mvdbeek — APPROVED** [review]
Looks good to me!

**Comment by DailyDreaming** [comment]
Also looks good to me for Toil.  We'll be adding support for this soon.

**Comment by GlassOfWhiskey** [comment]
[StreamFlow](https://github.com/alpha-unito/streamflow) passes all new Conformance tests. Nothing alarming in the CHANGELOG. For me, we can proceed.

File: CommandLineTool.yml [hunk 1]
```diff
@@ -37,10 +37,236 @@ $graph:
     - {$include: intro.md}
 
     - |
+      ## Introduction to the CWL Command Line Tool draft standard v1.2.1
```
Comment (tetron) [inline CommandLineTool.yml:40]: Since we're about to release it, we should remove the word "draft"

File: CommandLineTool.yml [hunk 2]
```diff
@@ -37,10 +37,236 @@ $graph:
...
+
+      This specification represents the latest development draft from the
+      CWL project. There are no new features nor behavior changes in CWL v1.2.1
+      as compared to CWL v1.2. v1.2.1 fixes only typos, adds clarifications,
+      and adds additional conformance tests. Some changes to the schema defining
+      CWL have been changed to aid the auto-generation of libraries for the reading
+      and writing of CWL documents.
+
+      Do not write `cwlVersion: v1.2.1`, nor `cwlVersion: v1.2.0`. The syntax
+      meaning of all terms in CWL `1.2.x is the same. However, when reporting
```
Comment (tetron) [inline CommandLineTool.yml:50]: This is slightly confusing, it should just say "Documents should continue to use `cwlVersion: v1.2`.

Comment (tetron) [reply CommandLineTool.yml:50]: also there's a dangling typo backtick "`1.2.x "

File: CommandLineTool.yml [hunk 3]
```diff
@@ -37,10 +37,236 @@ $graph:
     - {$include: intro.md}
 
     - |
+      ## Introduction to the CWL Command Line Tool draft standard v1.2.1
+
+      This specification represents the latest development draft from the
+      CWL project. There are no new features nor behavior changes in CWL v1.2.1
+      as compared to CWL v1.2. v1.2.1 fixes only typos, adds clarifications,
+      and adds additional conformance tests. Some changes to the schema defining
+      CWL have been changed to aid the auto-generation of libraries for the reading
```
Comment (tetron) [inline CommandLineTool.yml:46]: "Some changes ... have been changed"

File: CommandLineTool.yml [hunk 4]
```diff
@@ -37,10 +37,236 @@ $graph:
     - {$include: intro.md}
 
     - |
+      ## Introduction to the CWL Command Line Tool draft standard v1.2.1
+
+      This specification represents the latest development draft from the
```
Comment (tetron) [inline CommandLineTool.yml:42]: Remove "draft"

File: CommandLineTool.yml [hunk 5]
```diff
@@ -37,10 +37,236 @@ $graph:
...
+      CWL project. There are no new features nor behavior changes in CWL v1.2.1
+      as compared to CWL v1.2. v1.2.1 fixes only typos, adds clarifications,
```
Comment (tetron) [inline CommandLineTool.yml:44]: Should be "no new features compared to v1.2.0"

File: Workflow.yml [hunk 6]
```diff
@@ -34,6 +34,173 @@ $graph:
...
+      as compared to CWL v1.2. v1.2.1 fixes only typos, adds clarifications,
+      and adds additional conformance tests. Some changes to the schema defining
+      CWL have been changed to aid the auto-generation of libraries for the reading
+      and writing of CWL documents.
+
+      Do not write `cwlVersion: v1.2.1`, nor `cwlVersion: v1.2.0`. The syntax
+      meaning of all terms in CWL `1.2.x is the same. However, when reporting
+      results from running the CWL conformance tests, please do report all
+      three components; for example "99% of CWL v1.2.0 required tests" or
+      "100% of CWL v1.2.1 required tests".
```
Comment (tetron) [inline Workflow.yml:50]: See comments about front text in CommandLineTool

**Review by tetron — CHANGES_REQUESTED** [review]

**Comment by fmigneault** [comment]
Closes https://github.com/crim-ca/weaver/issues/549

**Review by tetron — DISMISSED** [review]

**Comment by stain** [comment]
All good from me :+1:

**Review by mvdbeek — APPROVED** [review]

**Comment by manabuishii** [comment]
Looks Good For Me

**Comment by tom-tan** [comment]
+1!

**Comment by GlassOfWhiskey** [comment]
Sorry for the delay. LGTM

**Comment by fabricebrito** [comment]
@mr-c working on calrissian conformance tests this week. Will keep you posted.

**Review by tetron — APPROVED** [review]

**Pull request merged**